addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2822,6 +2822,220 @@ options.registry.ReplaceImage = options.Class.extend({
     },
 });
 
+options.registry.ConditionalVisibility = options.Class.extend({
+    optionsAttributes: [
+        {
+            selectName: 'visibilityValueCountry',
+            dataName: 'data-country',
+        },
+        {
+            selectName: 'visibilityValueLang',
+            dataName: 'lang',
+        },
+        {
+            selectName: 'visibilityValueUtmMedium',
+            dataName: 'data-utm-medium',
+        },
+        {
+            selectName: 'visibilityValueUtmSource',
+            dataName: 'data-utm-source',
+        },
+        {
+            selectName: 'visibilityValueUtmCampaign',
+            dataName: 'data-utm-campaign',
+        },
+    ],
+    /**
+     * @constructor
+     */
+    init() {
+        this._super(...arguments);
+        if (this.$target[0].dataset.visibility === 'conditional') {
+            // hack to use the same visibility pannel as the popups
+            this.$target[0].classList.add('o_snippet_invisible');
```

Comment:
I think the o_snippet_invisible should be added the same time as when the conditionnal visibility is added, I don't see why this should be (and would work?) if in the "init" method.

Reply:
The issue is that at some point in the [cleanForSave](https://github.com/odoo/odoo/blob/af9454361f88e024af85d43204964ae396b7de9b/addons/web_editor/static/src/js/editor/snippets.editor.js#L290) of the snippets.edditor, if an element has an o_snippet_invisible it will set dataset.invisible and do a bunch of operations if it detects that o_snippet_invisible is on the target. That's why I remove it on the cleanForSave of the option as I think those operations are meant for popups and can lead to a block staying permanently hidden.

Reply:
After a better look into it, it doesn't seem like removing o_snippet_inivisible in cleanForSave prevents this from happening anyway. It doesn't seem like data-invisible impact the visibility.